### DISCUSSION TITLE
Not able to every package with pip install lakehouse_engine

### DISCUSSION THREAD

**Discussion by alexiska** [adidas/lakehouse-engine#12] [Q&A]
I am trying to pip install lakehouse_engine in Databricks and get following error:

```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
python-lsp-server 1.7.1 requires jedi<0.19.0,>=0.17.2, but you have jedi 0.19.1 which is incompatible.
databricks-sdk 0.1.6 requires requests<2.29.0,>=2.28.1, but you have requests 2.31.0 which is incompatible.
virtualenv 20.16.3 requires platformdirs<3,>=2.4, but you have platformdirs 3.11.0 which is incompatible.
```
Not sure if it is due to Databricks runtime version but I am using `13.3 LTS (includes Apache Spark 3.4.1, Scala 2.12)` at the moment.

Any idea how to solve this or is this package not compatible with Databricks?

**Comment by PMRocha** [comment]
Hello @alexiska ,
This library is compatible with Databricks and, when I tried to install using pip con a cluster running DBR 13.3 I get no error.
<img width="817" alt="image" src="https://github.com/adidas/lakehouse-engine/assets/5451606/df3eb59d-4a85-403f-95d8-0e09dd007b8c">
How are you making the installation?

**Reply by alexiska** [reply]
I am in a notebook and do a pip install 
![image](https://github.com/adidas/lakehouse-engine/assets/38601205/840592f4-8098-4933-8a5a-868b968a76b2)

**Reply by PMRocha** [reply]
Can you try and run this command?
<img width="817" alt="image" src="https://github.com/adidas/lakehouse-engine/assets/5451606/541b85b9-bd23-47f0-b1a0-b62c91f8f01e">
If you can you should be able to use the framework without issue ( we never noticed this error when using it in Databricks )

**Reply by alexiska** [reply]
I get this error when trying to import load_data:
![image](https://github.com/adidas/lakehouse-engine/assets/38601205/ed74b422-fccd-4db7-8bb6-9f997e41ab45)
This is due to the pip install of lakehouse_engine:
![image](https://github.com/adidas/lakehouse-engine/assets/38601205/1d76e56a-68a9-4823-9d25-d15f7dfae211)
 I will try to install the package in the cluster instead and see what I get

**Reply by PMRocha** [reply]
Do you have unity catalog enabled in your cluster?
If that is the case you wither have to restart the python interpreter ( using dbutils.library.restartPython() ) or restart your cluster to be able to use the functions.

**Reply by alexiska** [reply]
I installed the packae in the cluster instead of in the notebook and it works fine now.
Have I understood it correctly that this package is only working on AWS's S3 in the cloud?

**Reply by PMRocha** [reply]
Hello, 
the Lakehouse Engine was designed with S3 in mind but it should work for any backend except for some functions that were specifically designed for interacting with S3. The file manager is  currently working only with S3 and the default behavior of the data quality checks is to store information in an S3 bucket, you can however use your local file system.

If you face any problems please open an issue in the repo.

**Reply by alexiska** [reply]
That's unfortunate, I am using azure cloud storage

**Reply by PMRocha** [reply]
Sorry if I wasn't clear in my previous message. The way the framework is designed you should be able to use any backend that spark can read from/write to. In the case of data quality you can use local file system instead of an S3 bucket. Currently there is no alternative for the functions of the file manager.

**Reply by alexiska** [reply]
is there any plan to add the other cloud provider like azure?

**Reply by PMRocha** [reply]
Currently there are no plans to add the support, but feel free to contribute.
